src/nomad_measurements/xrd/schema.py

```diff
@@ -268,31 +299,34 @@ def derive_n_values(self):
             return len(self.two_theta)
         return 0
 
+    scan_type = Quantity(
+        type=str,
+        description='Type of scan (line, multiline, rsm)',
+    )
     n_values = Quantity(
         type=int,
         derived=derive_n_values,
     )
+    n_values_arr = Quantity(
+        type=np.dtype(np.int64),
+        shape=['*'],
+        derived=lambda self: np.arange(self.n_values),
+        description='Array of integers from 0 to n_values-1',
+    )
     two_theta = Quantity(
-        type=np.dtype(np.float64), shape=['n_values'],
+        type=np.dtype(np.float64), shape=['*'],
```

Comment:
If you do this, there is no need for `derive_n_values()`, `n_values`, or `n_values_arr` (can be removed)